### DISCUSSION TITLE
Getting distances on a route

### DISCUSSION THREAD

**Discussion by Sined66** [osmandapp/OsmAnd#17500] [Q&A] [answered]
Hi,
is it possible to have a distance from the starting point when selecting a POI on a route ?
If I select a petrol station near my route, details of the petrol station shows me kilometers from my starting point, but only in a straight line, not via my route.
I'd like to have the possibility to select any point ON my route and having the kilometers from my starting point.
Is it possible ? If so, how can I do that ?
Thanks in advance.
Denis

**Comment by mikehgentry** [comment, marked as answer]
If you swipe the navigation screen upwards then hit 'Details' and 'Analyse on map' you should be able to scroll around the graph that appears and have the corresponding point on the route line highlighted. Bit of an indirect way to do it, but sort of works

**Reply by Sined66** [reply]
Thanks for the workaround...
But not easy when route is hundreds kilometers length !

**Reply by mikehgentry** [reply]
I think it's a really good idea. Wouldn't need to be too sophisticated in my opinion (just anything within a small distance of the route, and ignoring later passes if you pass it several times).

I would suggest logging it as a feature request, or if you don't want to I will do so eventually.

**Reply by Sined66** [reply]
Yes, as Osmand is able to show any point on route when selecting it via the 'analyse on map', it should be able to do the reverse function !
